sink/sink

```diff
@@ -412,17 +417,62 @@ def mkdir_and_chdir(base, identifier):
     os.fchmod(dirfd, 0o755)
     os.close(dirfd)
 
+def prune_logs(base_path = ".", dry_run = False):
+    reference_time = time.time()
+
+    entries = os.listdir(base_path)
+    for entry in entries:
+        path = os.path.join(base_path, entry)
+        log_path = os.path.join(path, "log")
+
+        # we only care about directories with our "log" file in them
+        if not os.path.isdir(path) or not os.path.exists(log_path):
+            continue
+
+        # keep release logs
+        if entry.startswith("release-"):
+            continue
+
+        # use the last modified time of the logfile itself, not just the directory
+        last_modified = os.stat(log_path).st_mtime
+
+        diff_days = (reference_time - last_modified) / (60*60*24)
+
+        # keep anything newer than 30 days old
+        if diff_days < 30:
+            continue
+
+        archive_dir = os.path.join(base_path, "archive")
+        if not os.path.isdir(archive_dir):
+            try:
+                os.makedirs(archive_dir)
+            except:
+                pass
+
+        # keep the log summary, ignore nested backup folders
+        archive_file = os.path.join(archive_dir, "{0}-log.tar.xz".format(entry))
+        archive_command = ["tar",
+                           "cJf",
+                           archive_file,
+                           "--directory", base_path,
+                           os.path.join(entry, "log")]
+        # if the log has an html file for parsing, pack that as well for convenience
+        if os.path.isfile(os.path.join(path, "log.html")):
```

Comment:
Again this is specific to Cockpit, right? This is mentioned in the status file if we need to get it from somewhere more generic.